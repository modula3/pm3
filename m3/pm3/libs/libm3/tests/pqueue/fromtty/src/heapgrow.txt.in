showheap
i z
showheap
i y
showheap
i w
showheap
i x
showheap
i g
showheap
i f
showheap
i e
showheap
i d
showheap
i c
showheap
