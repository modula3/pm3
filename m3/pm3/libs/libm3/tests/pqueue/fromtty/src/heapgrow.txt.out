> showheap
LAST(pq.heap^) = 0; pq.sz = 0
pq.heap = Elts{NIL}
> i z
"z" => 1
> showheap
LAST(pq.heap^) = 1; pq.sz = 1
pq.heap = Elts{NIL, "z"}
> i y
"y" => 2
> showheap
LAST(pq.heap^) = 3; pq.sz = 2
pq.heap = Elts{NIL, "y", "z", NIL}
> i w
"w" => 3
> showheap
LAST(pq.heap^) = 3; pq.sz = 3
pq.heap = Elts{NIL, "w", "z", "y"}
> i x
"x" => 4
> showheap
LAST(pq.heap^) = 7; pq.sz = 4
pq.heap = Elts{NIL, "w", "x", "y", "z", NIL (X3)}
> i g
"g" => 5
> showheap
LAST(pq.heap^) = 7; pq.sz = 5
pq.heap = Elts{NIL, "g", "w", "y", "z", "x", NIL (X2)}
> i f
"f" => 6
> showheap
LAST(pq.heap^) = 7; pq.sz = 6
pq.heap = Elts{NIL, "f", "w", "g", "z", "x", "y", NIL}
> i e
"e" => 7
> showheap
LAST(pq.heap^) = 7; pq.sz = 7
pq.heap = Elts{NIL, "e", "w", "f", "z", "x", "y", "g"}
> i d
"d" => 8
> showheap
LAST(pq.heap^) = 15; pq.sz = 8
pq.heap = Elts{NIL, "d", "e", "f", "w", "x", "y", "g", "z", NIL (X7)}
> i c
"c" => 9
> showheap
LAST(pq.heap^) = 15; pq.sz = 9
pq.heap = Elts{NIL, "c", "d", "f", "e", "x", "y", "g", "z", "w", NIL (X6)}
> ^D
